wgengine/monitor/monitor.go

```diff
@@ -367,13 +364,28 @@ func (m *Mon) pollWallTime() {
 	m.wallTimer.Reset(pollWallTimeInterval)
 }
 
+func shouldMonitorTimeJump() bool {
+	switch runtime.GOOS {
+	case "android", "ios":
+		// For battery reasons, and because these platforms
+		// don't really sleep in the same way, don't poll
+		// for the wall time to detect for wake-for-sleep
+		// walltime jumps.
+		return false
+	default:
+		return true
+	}
+}
+
 // checkWallTimeAdvanceLocked updates m.timeJumped, if wall time jumped
 // more than 150% of pollWallTimeInterval, indicating we probably just
 // came out of sleep.
 func (m *Mon) checkWallTimeAdvanceLocked() {
 	now := wallTime()
-	if now.Sub(m.lastWall) > pollWallTimeInterval*3/2 {
-		m.timeJumped = true
+	if shouldMonitorTimeJump() {
```

Comment:
how about:

```go
if !shouldMonitorTimeJump {
    panic("unreachable") // if callers are correct
}
```

then fix the other caller.

Reply:
Done